ccmlib/node.py

```diff
@@ -1006,11 +1006,13 @@ def clear(self, clear_all=False, only_data=False, saved_caches=False):
             if only_data:
                 for dir in os.listdir(full_dir):
                     keyspace_dir = os.path.join(full_dir, dir)
-                    if os.path.isdir(keyspace_dir) and dir != "system":
+                    if os.path.isdir(keyspace_dir) and not dir.startswith("system"):
                         for f in os.listdir(keyspace_dir):
                             full_path = os.path.join(keyspace_dir, f)
                             if os.path.isfile(full_path):
                                 os.remove(full_path)
+                            elif d == "data":
+                                shutil.rmtree(full_path)
```

Comment:
why not just put something like
```py
for f in os.listdir(keyspace_dir):
    table_dir = os.path.join(keyspace_dir, f)
    shutil.rmtree(table_dir)
    os.mkdir(table_dir)
```

Reply:
You are right. For some reason I though we enter this path for commitlog and other mentioned directories but there is `if only_data:` So obviously we do not.

Reply:
Maybe using pathlib.Path could be a bit more fit and no need for 3 different modules to do this